ext-apiserver/cloud/google/pods.go

```diff
@@ -27,29 +29,81 @@ import (
 	"github.com/golang/glog"
 )
 
-var machineControllerImage = "gcr.io/k8s-cluster-api/machine-controller:0.19"
+var machineControllerImage = "gcr.io/k8s-cluster-api/apiserver-controller:0.1"
 
 func init() {
 	if img, ok := os.LookupEnv("MACHINE_CONTROLLER_IMAGE"); ok {
 		machineControllerImage = img
 	}
 }
 
+type caCertParams struct {
+	caBundle string
+	tlsCrt   string
+	tlsKey   string
+}
+
+func getBase64(file string) string {
+	buff := bytes.Buffer{}
+	enc := base64.NewEncoder(base64.StdEncoding, &buff)
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		glog.Fatalf("Could not read file %s: %v", file, err)
+	}
+
+	_, err = enc.Write(data)
+	if err != nil {
+		glog.Fatalf("Could not write bytes: %v", err)
+	}
+	enc.Close()
+	return buff.String()
+}
+
+func getApiServerCerts() (*caCertParams, error) {
+	os.RemoveAll("./config")
+	err := run("apiserver-boot", "build", "config",
```

Comment:
We should generate the certs without relying on apiserver-boot. You shouldn't need to install that to deploy a cluster.

Reply:
Opened https://github.com/kubernetes/kube-deploy/issues/591

Reply:
Can you take a look at this sample program and see if it would be easy enough to do it in this PR? https://golang.org/src/crypto/tls/generate_cert.go

Also, do we need the CA private key for anything? It seems like we would.

Reply:
This is rather big. I prefer to have someone else to work on it as I might not have enough bandwidth along the road. We need to make this big PR in so that people can make contribution in parallel.